sycl/test/invoke_simd/invoke_simd.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -c -fsycl %s
+// RUN: %clangxx -c -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr %s
```

Comment:
Why was fsycl-allow-func-ptr not needed previously? Can we suppress the FE error just for our header without allowing it for user code?

Reply:
Previously, FE somehow allowed `invoke_simd(func,...)` before, even though `first parameter of invoke_simd is a function pointer, I think we were just lucky. With this change, user function is explicitly called, so now FE correctly complains, and `-fsycl-allow-func-ptr` is expectedly needed.

Since this is still experimental, I think we can start with requiring the extra option, and then after some maturing fix the FE (which is pretty lengthy process)